src/reflect/gclayout_precise.go

```diff
@@ -0,0 +1,37 @@
+//go:build gc.precise
+
+package reflect
+
+import "unsafe"
+
+var (
+	// Constants for use with alloc()
+	// See runtime/gc_precise.go
+	// pppppppp_pppppppp_pppppppp_ppsssss1
+	gcLayoutNoPtrs  = unsafe.Pointer(uintptr(0b0000_00001_1))
+	gcLayoutPointer = unsafe.Pointer(uintptr(0b0001_00001_1))
+	gcLayoutString  = unsafe.Pointer(uintptr(0b0001_00010_1))
+	gcLayoutSlice   = unsafe.Pointer(uintptr(0b0001_00011_1))
+)
```

Comment:
Unfortunately all these constants (specifically the amount of bits used for the size of the object) vary by pointer size (16, 32, or 64 bits).
I think it may be possible to get the right constants with some const bit twiddling?
Much easier would probably be to write a function that calculates these at runtime (switching over `unsafe.Sizeof(uintptr(0))`) and hope the function gets inlined and optimized to a single constant.